modules/reactor/doc/content/source/meshgenerators/PatternedHexMeshGenerator.md

```diff
@@ -0,0 +1,50 @@
+# PatternedHexMeshGenerator
+
+!syntax description /Mesh/PatternedHexMeshGenerator
+
+## Overview
+
+The `PatternedHexMeshGenerator` object generates a 2D mesh by stitching hexagonal meshes generated by [`PolygonConcentricCircleMeshGenerator`](/PolygonConcentricCircleMeshGenerator.md), [`HexagonConcentricCircleAdaptiveBoundaryMeshGenerator`](/HexagonConcentricCircleAdaptiveBoundaryMeshGenerator.md), and itself, based on a user-defined hexagonal grid pattern. The generated 2D mesh can optionally contain an extra background region and/or external duct regions to form a hexagonal external shape rather than a jagged boundary edge.
+
+## Geometry Information
+
+In order to generate the hexagonal patterned mesh, three fundamental parameters are needed:
+
+- `inputs`: a vector of mesh generator names that will be used as elements to form the patterned mesh;
+- `pattern`: a two-dimensional vector that represents the pattern of mesh to be generated. The elements must be integers from 0 to N-1, where N is the length of vector parameter `inputs`.
+- `pattern_boundary`: a MooseEnum parameter that can be either `hexagon` or `none`. When `none` is selected, this object only stitches `inputs` meshes into a patterned mesh without extraneous geometry, resulting in an outer boundary with a zig-zag edge. When `hexagon` is selected, a background region is added so that the generated mesh has a hexagonal shape instead of a zig-zag one. Concentric duct regions may also be optionally added to the hexagon periphery when this parameter is `hexagon`. The user can refer to [Figure 1](#pattern_hex) for more details.
```

Comment:
```suggestion
- [!param](/Mesh/PatternedHexMeshGenerator/pattern_boundary): a MooseEnum parameter that can be either `hexagon` or `none`. When `none` is selected, this object only stitches [!param](/Mesh/PatternedHexMeshGenerator/inputs( meshes into a patterned mesh without extraneous geometry, resulting in an outer boundary with a zig-zag edge. When `hexagon` is selected, a background region is added so that the generated mesh has a hexagonal shape instead of a zig-zag one. Concentric duct regions may also be optionally added to the hexagon periphery when this parameter is `hexagon`. The user can refer to [Figure 1](#pattern_hex) for more details.
```